pkg/kube/secretcontroller/secretcontroller.go

```diff
@@ -219,22 +219,27 @@ func (c *Controller) Run(stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
 	defer c.queue.ShutDown()
 
+	t0 := time.Now()
 	log.Info("Starting Secrets controller")
 
 	go c.informer.Run(stopCh)
 
-	// Wait for the caches to be synced before starting workers
-	if !kube.WaitForCacheSyncInterval(stopCh, c.syncInterval, c.informer.HasSynced) {
-		return
-	}
-	// all secret events before this signal must be processed before we're marked "ready"
-	c.queue.Add(initialSyncSignal)
-	if features.RemoteClusterTimeout != 0 {
-		time.AfterFunc(features.RemoteClusterTimeout, func() {
-			c.remoteSyncTimeout.Store(true)
-		})
-	}
-	go wait.Until(c.runWorker, 5*time.Second, stopCh)
+	go func() {
+		if !cache.WaitForCacheSync(stopCh, c.informer.HasSynced) {
+			log.Error("Failed to sync secret controller cache")
+			return
+		}
+		log.Info("Secret controller synced ", time.Since(t0))
```

Comment:
nit: maybe should be after we add all events?

Reply:
It could be useful to differentiate the point where caches have synced, and we've finished processing our queue.